x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx

```diff
@@ -377,6 +397,50 @@ const AlertsTable: React.FunctionComponent<AlertsTableProps> = (props: AlertsTab
     return props.columns;
   }, [getCellActions, disabledCellActions, props.columns, visibleCellActions]);
 
+  // Merges the default grid style with the grid style that comes in through props.
+  const actualGridStyle = useMemo(() => {
+    const propGridStyle: NonNullable<EuiDataGridStyle> = props.gridStyle ?? {};
+    // Merges default row classes, custom ones and adds the active row class style
+    const mergedGridStyle: EuiDataGridStyle = {
+      ...DefaultGridStyle,
+      ...propGridStyle,
+      rowClasses: {
+        // We're spreadind the highlighted row classes first, so that the active
+        // row classed can override the highlighted row classes.
+        ...highlightedRowClasses,
+        ...activeRowClasses,
+      },
+    };
+
+    // If ANY additional rowClasses have been provided, we need to merge them with our internal ones
+    if (propGridStyle.rowClasses) {
+      // Get all row indices with a rowClass.
+      const mergedKeys = [
+        ...Object.keys(activeRowClasses),
+        ...Object.keys(propGridStyle.rowClasses || {}),
+      ];
+      // Deduplicate keys to avoid extra iterations
+      const dedupedKeys = Array.from(new Set(mergedKeys));
+
+      // For each index, merge row classes
+      const mergedRowClasses = dedupedKeys.reduce<NonNullable<EuiDataGridStyle['rowClasses']>>(
+        (rowClasses, key) => {
+          const intKey = parseInt(key, 10);
+
+          // Concatenate class names for active row class and passed-in row class
+          // (if either exists)
+          rowClasses[intKey] = [activeRowClasses[intKey], propGridStyle.rowClasses?.[intKey]]
```

Comment:
Should it also contain the `highlightedRowClasses`?  Are we saying that if `propGridStyle.rowClasses` is passed, `highlightedRowClasses` will not be taking into consideration?

Reply:
Yep.. Even if I give rowClass for only one row in `propsGridStyle`, it ignore highlighted rowClasses entirely. Here is the video.

https://user-images.githubusercontent.com/7485038/234807272-2eac0752-1f60-4b41-bfd6-e065c492a85e.mov

Reply:
Nice catch, fixed in https://github.com/elastic/kibana/pull/155497/commits/fed125a6e41b763562df8b5432e300f9013869d0 and https://github.com/elastic/kibana/pull/155497/commits/77ea66dd74f842198133a067c21f4ca9a1e1de45.

It makes an assumption though that internal classes have precedence over custom ones. That made sense to me, but maybe @jcger has a different opinion?

Reply:
I agree with that.